wic/iot2050-swu.wks

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) Siemens AG, 2021
+#
+# Authors:
+#  Quirin Gylstorff <quirin.gylstorff@siemens.com>
+#
+# This file is subject to the terms and conditions of the MIT License.  See
+# COPYING.MIT file in the top-level directory.
+#
+
+bootloader --ptable gpt --append "console=ttyS3,115200n8 earlycon=ns16550a,mmio32,0x02810000 mtdparts=47040000.spi.0:512k(ospi.tiboot3),2m(ospi.tispl),4m(ospi.u-boot),128k(ospi.env),128k(ospi.env.backup),1m(ospi.sysfw),64k(pru0-fw),64k(pru1-fw),64k(rtu0-fw),64k(rtu1-fw),-@8m(ospi.rootfs) rw rootwait"
+
+part --source rootfs-u-boot --sourceparams="no_initrd=yes,script_prepend=if test \\\"\${board_name}\\\" = \\\"IOT2050-ADVANCED\\\"; then set fdtfile ti/k3-am6548-iot2050-advanced-oldfw.dtb; else setenv fdtfile ti/k3-am6528-iot2050-basic-oldfw.dtb; fi;env exists sysselect || setenv sysselect 1,root=PARTUUID:fedcba98-7654-3210-cafe-5e071000000\${sysselect}" --fstype ext4 --label rootfsa --align 1024 --uuid "fedcba98-7654-3210-cafe-5e0710000001"
+
+part --source rootfs-u-boot --sourceparams="no_initrd=yes,script_prepend=if test \\\"\${board_name}\\\" = \\\"IOT2050-ADVANCED\\\"; then set fdtfile ti/k3-am6548-iot2050-advanced-oldfw.dtb; else setenv fdtfile ti/k3-am6528-iot2050-basic-oldfw.dtb; fi;env exists sysselect || setenv sysselect 1,root=PARTUUID:fedcba98-7654-3210-cafe-5e071000000\${sysselect}" --fstype ext4 --label rootfsb --align 1024 --uuid "fedcba98-7654-3210-cafe-5e0710000002"
```

Comment:
How about a persistent data partition?

Reply:
The necessary wic plugin is currently missing upstream.

Reply:
Which one?

I was comparing this to https://github.com/siemens/meta-iot2000/blob/master/meta-iot2000-example/scripts/lib/wic/canned-wks/example-image.iot2000.swupdate.wks where we had a data partition (but for yocto).